zilencer/management/commands/populate_db.py

```diff
@@ -370,6 +371,16 @@ def handle(self, **options):
             # Mark all messages as read
             UserMessage.objects.all().update(flags=UserMessage.flags.read)
 
+            if not options["test_suite"]:
+                # Update pointer of each user to point to the last message in their
+                # UserMessage rows with sender_id=user_profile_id.
+                users = list(UserMessage.objects.filter(
+                    message__sender_id=F('user_profile_id')).values(
+                    'user_profile_id').annotate(pointer=Max('message_id')))
+                for user in users:
+                    UserProfile.objects.filter(id=user['user_profile_id']).update(
+                        pointer=user['pointer'])
```

Comment:
Ideally, we'd do this with a single bulk query that updates all of the users.  That would help make this script as fast as possible, especially when using the `--extra-users` option.  I'm going to merge for now, but would be a good follow-up to change that.